corehq/form_processor/models/cases.py

```diff
@@ -279,15 +279,27 @@ def soft_undelete_cases(self, domain, case_ids):
             publish_case_saved(case)
         return undeleted_count
 
-    def hard_delete_cases(self, domain, case_ids, *, publish_changes=True):
+    def hard_delete_cases(self, domain, case_ids, *, publish_changes=True, leave_tombstone=False):
```

Comment:
I think we should set the default to `True` and change all the places that currently use the function to explicitly pass `leave_tombstone=False`. That way a new usage that forgets to pass this argument will quickly find out that tombstones need to be considered.